src/System.IO.Compression/src/System/IO/Compression/Crc32Helper.cs

```diff
@@ -449,11 +453,40 @@ internal static class Crc32Helper
             0x6EAB0882u, 0xA201081Cu, 0xA8C40105u, 0x646E019Bu, 0xEAE10678u,
             0x264B06E6u
         };
-
+        private static WorkerType GetCrcType()
+        {
+            try
+            {
+                // Make any P/Invoke into zlib to ensure we're able to find and use it.
+                // If we are, then use zlib.
+                Interop.zlib.zlibCompileFlags();
```

Comment:
Seems we should actually try to invoke crc32, so that we can use it if the export is available and not if it's not.  I'm concerned we could find ourselves in a situation where someone's using this code with a clrcompression.dll that doesn't have the export, as is apparently currently the case with the version being used in our tests (which we should separately look into to understand why it wasn't updated).